### PR TITLE
Switched to Custom War Packager build

### PR DESCRIPTION
**Pull request by alxsap**
Introduced Custom War Packager and Configuration-as-Code (casc).
This way we can build more sophisticated images.
For example the image now contains the Kubernetes plugin
with prepared configuration to run Pipeline steps inside
docker containers on the same Kubernetes cluster.

### CODE REVIEW

File: jenkinsfile-runner-base-image/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+function die() {
+  if (( ${#@} > 0 )); then
+    echo "$@" >&2
+  fi
+  echo
+  echo "FAILED"
+  exit 1
+}
+
```
Comment (anbrsap) [inline jenkinsfile-runner-base-image/build.sh]: Add this to become independent of the caller's current directory:

```bash
cd "$(dirname "$BASH_SOURCE")"
```

File: jenkinsfile-runner-steward-image/build.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+function die() {
+  if (( ${#@} > 0 )); then
+    echo "$@" >&2
+  fi
+  echo
+  echo "FAILED"
+  exit 1
+}
+
```
Comment (anbrsap) [inline jenkinsfile-runner-steward-image/build.sh]: Dito:

```bash
cd "$(dirname "$BASH_SOURCE")"
```

File: jenkinsfile-runner-steward-image/scripts/run.sh [hunk 3]
```diff
@@ -118,21 +122,40 @@ function random_alnum() {
   echo "${chars:0:$len}"
 }
 
+function configure_log_elasticsearch() {
+  with_termination_log python3 -b -B -E -I "${HERE}/create_elasticsearch_log_config.py" > "${_JENKINS_CASC_D}/log-elasticsearch.yml" || return 1
+  # TODO remove debug code
+  { cat "${_JENKINS_CASC_D}/log-elasticsearch.yml"; echo; } || true
```
Comment (anbrsap) [inline jenkinsfile-runner-steward-image/scripts/run.sh]: Shouldn't we remove this?

File: build.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+function die() {
+  if (( ${#@} > 0 )); then
+    echo "$@" >&2
+  fi
+  echo
+  echo "FAILED"
+  exit 1
+}
+
+PROJECT_ROOT=${PWD}
```
[the remark this answers is not in the document]
Comment (alxsap) [reply build.sh]: Will the quotes work like this?

**Pull request merged**